### PR TITLE
`FeatureFormView` - Attachment handling revisions

### PR DESCRIPTION
**Pull request by dfeinzimer**
Related Apollo 684

### CODE REVIEW

**Comment by dfeinzimer** [comment]
This is ready for review once again

**Comment by dfeinzimer** [comment]
Still ready for review. Updated `AttachmentFormElement.deleteAttachment(attachment)` to `AttachmentFormElement.delete(attachment)`.

**Comment by mhdostal** [comment]
> Updated `AttachmentFormElement.deleteAttachment(attachment)` to `AttachmentFormElement.delete(attachment)`.

Is that what the new generated code outputs?

File: Sources/ArcGISToolkit/Common/AttachmentsFeatureElementView.swift [hunk 1]
```diff
@@ -177,8 +177,7 @@ struct AttachmentsFeatureElementView: View {
     ///   - newAttachmentName: The new attachment name.
     @MainActor
     func onRename(attachmentModel: AttachmentModel, newAttachmentName: String) async throws -> Void {
```
Comment (mhdostal) [inline Sources/ArcGISToolkit/Common/AttachmentsFeatureElementView.swift:179]: This no longer needs to be `async throws`.
```suggestion
    func onRename(attachmentModel: AttachmentModel, newAttachmentName: String) -> Void {
```

File: Sources/ArcGISToolkit/Common/AttachmentsFeatureElementView.swift [hunk 2]
```diff
@@ -192,10 +191,10 @@ struct AttachmentsFeatureElementView: View {
     func onDelete(attachmentModel: AttachmentModel) async throws -> Void {
```
Comment (mhdostal) [inline Sources/ArcGISToolkit/Common/AttachmentsFeatureElementView.swift:191]: This no longer needs to be `async throws`.
```suggestion
    func onDelete(attachmentModel: AttachmentModel) -> Void {
```

**Comment by mhdostal** [comment]
Does the `AttachmentImportMenu.init(element: AttachmentsFormElement, onAdd: ((FeatureAttachment) async throws -> Void)? = nil)` method still need the `onAdd` argument to be `async throws`?

Same question for `onDelete` and `onRename` for the `AttachmentPreview` class?

**Review by mhdostal — CHANGES_REQUESTED** [review]

**Comment by dfeinzimer** [comment]
> Is that what the new generated code outputs?

It looks like taking the `attachments` property out of generation to make it not-cached actually caused that.. which I wouldn't have expected. Reverting for now.

**Comment by mhdostal** [comment]
I think `AttachmentFormElement.delete(attachment)` is actually better. See the Swift API doc [here](https://www.swift.org/documentation/api-design-guidelines/#naming).

**Review by mhdostal — APPROVED** [review]
This looks good. We can figure out the naming during the API Audit.

**Pull request merged**